src/event/event-update.ts

```diff
@@ -0,0 +1,25 @@
+import {Change} from 'firebase-functions';
+import {QueryDocumentSnapshot} from 'firebase-functions/lib/providers/firestore';
+import {sheetService} from '../sheets/sheet.service';
+import {EventModel} from '../core/models/event.model';
+
+export const onEventUpdated = (snapshot: Change<QueryDocumentSnapshot>, context: any) => {
+    const oldEvent = snapshot.before.data() as EventModel;
+    const newEvent = snapshot.after.data() as EventModel;
+    if (
+        oldEvent.course !== newEvent.course ||
+        oldEvent.chapter !== newEvent.chapter ||
+        oldEvent.university !== newEvent.university ||
+        oldEvent.startDate !== newEvent.startDate ||
+        oldEvent.endDate !== newEvent.endDate ||
+        oldEvent.relatedFiles !== newEvent.relatedFiles ||
+        oldEvent.recorder !== newEvent.recorder ||
+        oldEvent.transcripter !== newEvent.transcripter ||
+        oldEvent.teacher !== newEvent.teacher ||
+        oldEvent.sheetMaker !== newEvent.sheetMaker ||
+        oldEvent.onlineCourse !== newEvent.onlineCourse ||
+        oldEvent.courseLink !== newEvent.courseLink
```

Comment:
Have a look at the lodash lib. There might be a function that does exactly what you want here (deep compare)

Reply:
OK, so there is the _isEqual function, but it compares everything in both objects when I only want to compare the selected attributes... Maybe I am comparing so many attributes that it doesn't matter.
If ever we don't care about only updating sheets when these specific attributes are edited on events, maybe the 'if' statement should be removed altogether: the method is called only when events are edited after all.

Reply:
We could do it that way https://stackoverflow.com/questions/37922967/exclude-some-properties-in-comparison-using-isequal-of-lodash

Reply:
@AntoineXevlabs OK, thanks to your comment I've finally found the one I was searching: _.pick, which is apparently faster than _.omit (according to https://docs-lodash.com/v4/omit/ ). So I'll probably use _.pick even if it's longer to write.

Reply:
Hum, code cleaning is better than time processing here (we'll have ms differences). Let's use omit since we have less key that we want to throw

Reply:
OK